File: framework/ui/src/io/intino/alexandria/ui/displays/components/SetDarkMode.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.events.actionable.ExecuteEvent;
import io.intino.alexandria.ui.displays.events.actionable.ExecuteListener;
import io.intino.alexandria.ui.displays.notifiers.SetDarkModeNotifier;

public class SetDarkMode<DN extends SetDarkModeNotifier, B extends Box> extends AbstractSetDarkMode<DN, B> {
	private ExecuteListener executeListener;

	public SetDarkMode(B box) {
		super(box);
	}

	public SetDarkMode<DN, B> onExecute(ExecuteListener listener) {
		this.executeListener = listener;
		return this;
	}

	public void execute() {
		notifier.updateMode();
		soul().displays(SetLightMode.class).forEach(d -> d.visible(true));
		soul().displays(SetDarkMode.class).forEach(d -> d.visible(false));
		executeListener.accept(new ExecuteEvent(this, "dark"));
	}

}

File: framework/ui-elements/src/displays/components/SetDarkMode.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractSetDarkMode from "../../../gen/displays/components/AbstractSetDarkMode";
import SetDarkModeNotifier from "../../../gen/displays/notifiers/SetDarkModeNotifier";
import SetDarkModeRequester from "../../../gen/displays/requesters/SetDarkModeRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import { withSnackbar } from 'notistack';
import Theme from 'app-elements/gen/Theme';
import Actionable from "./Actionable";

class SetDarkMode extends AbstractSetDarkMode {

	constructor(props) {
		super(props);
		const theme = Theme.get();
		this.notifier = new SetDarkModeNotifier(this);
		this.requester = new SetDarkModeRequester(this);
		this.state = {
            ...this.state,
            visible: this.state.appMode == "light"
		};
	};

	render() {
	    return (
	        <React.Fragment>
	            {this.renderCookieConsent()}
	            {super.render()}
	        </React.Fragment>
	    );
	};

	updateMode = () => {
	    Theme.get().setMode("dark");
	    this._saveAppModeInCookies("dark");
	    this.setState({appMode: mode});
	};

}

export default withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SetDarkMode));
DisplayFactory.register("SetDarkMode", withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SetDarkMode)));